tool/tsh/tsh.go

```diff
@@ -1898,6 +1899,63 @@ func onListNodes(cf *CLIConf) error {
 	return nil
 }
 
+// clusterClient is a client for a particular cluster
+type clusterClient struct {
+	name    string
+	tc      *client.TeleportClient
```

Comment:
Nit: I'm not an expert here but exposing `TeleportClient` within `clusterClient` is slightly risky to me. `TeleportClient` is not safe for concurrent use while `clusterClient` was made specifically for use in a concurrent context. We could get away with exposing just the return value of `tc.DefaultResourceFilter`, couldn't we?